From 2b079fa099dff5ca164c6e9be6ac97e862e0029c Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Thu, 23 Nov 2017 14:51:14 +0200
Subject: [PATCH 145/767] staging: fsl-dpaa2/eth: Configure HWA on XDP_TX FD

Explicitly reserve a hardware annotation area of 128B in the XDP_TX FD,
since we write to the HWA[FAEAD] field to disable Tx confirmations.

We don't actually use the software annotation area, but since we reuse
the Rx FD and buffer, let the hardware know the SWA is reserved.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index 4c5aafe62cf3..14da1cbda147 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -247,6 +247,7 @@ static int dpaa2_eth_xdp_tx(struct dpaa2_eth_priv *priv,
 	/* Mark the egress frame annotation area as valid */
 	frc = dpaa2_fd_get_frc(fd);
 	dpaa2_fd_set_frc(fd, frc | DPAA2_FD_FRC_FAEADV);
+	dpaa2_fd_set_ctrl(fd, DPAA2_FD_CTRL_PTA | DPAA2_FD_CTRL_ASAL);
 
 	ctrl = DPAA2_FAEAD_A4V | DPAA2_FAEAD_A2V | DPAA2_FAEAD_EBDDV;
 	faead = dpaa2_get_faead(buf_start);
-- 
2.17.0

